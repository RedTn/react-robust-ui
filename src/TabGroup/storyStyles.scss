.setFixedWidth{
  width: 600px;
}

.icon {
  width: 20px;
  height: 20px;
}

.icon-green-blur {
  width: 20px;
  height: 20px;
  fill: #50B83C;
}

.vertical-center {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
