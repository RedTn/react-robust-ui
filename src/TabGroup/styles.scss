$underline-size: 5px;

.tab-group-article{
  display: flex;
  justify-content: space-between;
}

.pointer{
  cursor: pointer;
}

.underline{
  border-bottom: $underline-size solid teal;
}

.underline-hidden{
  border-bottom: $underline-size solid transparent;
}

.tab-padding{
  padding: 20px;
}

.flex {
  display: flex;
  justify-content: space-between;
}

.tab-group-ul{
  margin: 0;
}

.tab-li{
  list-style-type: none;
}

.divider-padding {
  padding: 20px 10px 20px 10px;
}

.no-margin{
  margin: 0;
}

.text-center {
  text-align: center;
}

.inline-block {
  display: inline-block;
}
