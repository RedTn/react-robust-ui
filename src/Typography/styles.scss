.button{
    height: 19px;
    font-family: Lato;
    font-size: 16px;
    font-weight: bold;
}