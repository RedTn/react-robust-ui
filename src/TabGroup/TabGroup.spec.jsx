import React from 'react';
import TabGroup from './TabGroup';
import Typography from '../Typography/Typography.jsx';
import { shallow } from 'enzyme';

describe('TabGroup', () => {
    describe('integration', () => {
        it('matchs base snapshot', () => {
            expect.assertions(1);

            const wrapper = shallow(<TabGroup />);
            expect(wrapper).toMatchSnapshot();
        });
        it('matches basic usage snapshot', () => {
            expect.assertions(1);

            const wrapper = shallow(
                <TabGroup>
                    <TabGroup.Tab value={0}>Test</TabGroup.Tab>
                    <TabGroup.Tab defaultActive value={1}>
            Test2
                    </TabGroup.Tab>
                    <TabGroup.Tab value={2}>Test3</TabGroup.Tab>
                </TabGroup>
            );
            expect(wrapper).toMatchSnapshot();
        });
        it('renders basic element if not a tab', () => {
            expect.assertions(1);

            const wrapper = shallow(
                <TabGroup>
                    <TabGroup.Tab value={0}>Test</TabGroup.Tab>
                    <TabGroup.Tab defaultActive value={1}>
            Test2
                    </TabGroup.Tab>
                    <span>This is not a tab</span>
                </TabGroup>
            );
            expect(wrapper).toMatchSnapshot();
        });
        it('makes clicked tab the active one', () => {
            expect.assertions(4);

            const callback = jest.fn();
            const wrapper = shallow(
                <TabGroup onChange={callback}>
                    <TabGroup.Tab value={0}>Test</TabGroup.Tab>
                    <TabGroup.Tab value={1}>Test2</TabGroup.Tab>
                    <span>This is not a tab</span>
                </TabGroup>
            );
            let firstTab = wrapper.children().filterWhere(n => n.prop('value') === 0);
            expect(firstTab).toMatchSnapshot();
            expect(callback).not.toHaveBeenCalled();
            firstTab.prop('selectTab')();
            wrapper.update();
            firstTab = wrapper.children().filterWhere(n => n.prop('value') === 0);
            expect(callback).toHaveBeenCalled();
            expect(firstTab).toMatchSnapshot();
        });
    });
    describe('unit', () => {});
    describe('Tab', () => {
        describe('integration', () => {
            it('is not active be default', () => {
                expect.assertions(2);

                const wrapper = shallow(<TabGroup.Tab />);
                expect(wrapper.find(Typography).prop('variant')).toBe('tab');
                expect(wrapper).toMatchSnapshot();
            });
            it('has active class typography when active', () => {
                expect.assertions(2);

                const wrapper = shallow(<TabGroup.Tab isActive />);
                expect(wrapper.find(Typography).prop('variant')).toBe('tab-active');
                expect(wrapper).toMatchSnapshot();
            });
            it('bubbles selectTab when clicked', () => {
                expect.assertions(2);

                const callback = jest.fn();
                const wrapper = shallow(<TabGroup.Tab selectTab={callback} />);
                expect(callback).not.toHaveBeenCalled();
                wrapper.find('span').prop('onClick')();
                expect(callback).toHaveBeenCalled();
            });
        });
        describe('unit', () => {});
    });
});
