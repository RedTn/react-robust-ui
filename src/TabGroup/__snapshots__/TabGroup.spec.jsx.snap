// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabGroup Tab integration has active class typography when active 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    isActive={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Typography
        className="no-margin"
        inline={false}
        style={null}
        variant="tab-active"
      />,
      "className": "underline pointer tab-padding",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": null,
        "className": "no-margin",
        "inline": false,
        "style": null,
        "variant": "tab-active",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Typography
          className="no-margin"
          inline={false}
          style={null}
          variant="tab-active"
        />,
        "className": "underline pointer tab-padding",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "className": "no-margin",
          "inline": false,
          "style": null,
          "variant": "tab-active",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TabGroup Tab integration is not active be default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Typography
        className="no-margin"
        inline={false}
        style={null}
        variant="tab"
      />,
      "className": "underline-hidden pointer tab-padding",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": null,
        "className": "no-margin",
        "inline": false,
        "style": null,
        "variant": "tab",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Typography
          className="no-margin"
          inline={false}
          style={null}
          variant="tab"
        />,
        "className": "underline-hidden pointer tab-padding",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "className": "no-margin",
          "inline": false,
          "style": null,
          "variant": "tab",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TabGroup integration makes clicked tab the active one 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <TabGroup
      className={null}
      defaultValue={null}
      onChange={[MockFunction]}
      style={null}
    >
      <Unknown
        value={0}
      >
        Test
      </Unknown>
      <Unknown
        value={1}
      >
        Test2
      </Unknown>
      <span>
        This is not a tab
      </span>
    </TabGroup>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <span>
            This is not a tab
          </span>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": false,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "isActive": false,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "host",
          "props": Object {
            "children": "This is not a tab",
          },
          "ref": null,
          "rendered": "This is not a tab",
          "type": "span",
        },
      ],
      "type": "article",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              isActive={false}
              selectTab={[Function]}
              value={0}
            >
              Test
            </Unknown>,
            <Unknown
              isActive={false}
              selectTab={[Function]}
              value={1}
            >
              Test2
            </Unknown>,
            <span>
              This is not a tab
            </span>,
          ],
          "className": "tab-group-article",
          "style": null,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": ".0",
            "nodeType": "function",
            "props": Object {
              "children": "Test",
              "isActive": false,
              "selectTab": [Function],
              "value": 0,
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".1",
            "nodeType": "function",
            "props": Object {
              "children": "Test2",
              "isActive": false,
              "selectTab": [Function],
              "value": 1,
            },
            "ref": null,
            "rendered": "Test2",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".2",
            "nodeType": "host",
            "props": Object {
              "children": "This is not a tab",
            },
            "ref": null,
            "rendered": "This is not a tab",
            "type": "span",
          },
        ],
        "type": "article",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": true,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": ".0",
    "nodeType": "function",
    "props": Object {
      "children": "Test",
      "isActive": false,
      "selectTab": [Function],
      "value": 0,
    },
    "ref": null,
    "rendered": "Test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": ".0",
      "nodeType": "function",
      "props": Object {
        "children": "Test",
        "isActive": false,
        "selectTab": [Function],
        "value": 0,
      },
      "ref": null,
      "rendered": "Test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <span>
            This is not a tab
          </span>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": false,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "isActive": false,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "host",
          "props": Object {
            "children": "This is not a tab",
          },
          "ref": null,
          "rendered": "This is not a tab",
          "type": "span",
        },
      ],
      "type": "article",
    },
  ],
}
`;

exports[`TabGroup integration makes clicked tab the active one 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <TabGroup
      className={null}
      defaultValue={null}
      onChange={
        [MockFunction] {
          "calls": Array [
            Array [
              0,
              <Unknown
                value={0}
              >
                Test
              </Unknown>,
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
      style={null}
    >
      <Unknown
        value={0}
      >
        Test
      </Unknown>
      <Unknown
        value={1}
      >
        Test2
      </Unknown>
      <span>
        This is not a tab
      </span>
    </TabGroup>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={true}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <span>
            This is not a tab
          </span>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": true,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "isActive": false,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "host",
          "props": Object {
            "children": "This is not a tab",
          },
          "ref": null,
          "rendered": "This is not a tab",
          "type": "span",
        },
      ],
      "type": "article",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              isActive={true}
              selectTab={[Function]}
              value={0}
            >
              Test
            </Unknown>,
            <Unknown
              isActive={false}
              selectTab={[Function]}
              value={1}
            >
              Test2
            </Unknown>,
            <span>
              This is not a tab
            </span>,
          ],
          "className": "tab-group-article",
          "style": null,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": ".0",
            "nodeType": "function",
            "props": Object {
              "children": "Test",
              "isActive": true,
              "selectTab": [Function],
              "value": 0,
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".1",
            "nodeType": "function",
            "props": Object {
              "children": "Test2",
              "isActive": false,
              "selectTab": [Function],
              "value": 1,
            },
            "ref": null,
            "rendered": "Test2",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": ".2",
            "nodeType": "host",
            "props": Object {
              "children": "This is not a tab",
            },
            "ref": null,
            "rendered": "This is not a tab",
            "type": "span",
          },
        ],
        "type": "article",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": true,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": ".0",
    "nodeType": "function",
    "props": Object {
      "children": "Test",
      "isActive": true,
      "selectTab": [Function],
      "value": 0,
    },
    "ref": null,
    "rendered": "Test",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": ".0",
      "nodeType": "function",
      "props": Object {
        "children": "Test",
        "isActive": true,
        "selectTab": [Function],
        "value": 0,
      },
      "ref": null,
      "rendered": "Test",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={true}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <span>
            This is not a tab
          </span>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": true,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "isActive": false,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "host",
          "props": Object {
            "children": "This is not a tab",
          },
          "ref": null,
          "rendered": "This is not a tab",
          "type": "span",
        },
      ],
      "type": "article",
    },
  ],
}
`;

exports[`TabGroup integration matches basic usage snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabGroup
    className={null}
    defaultValue={null}
    onChange={null}
    style={null}
  >
    <Unknown
      value={0}
    >
      Test
    </Unknown>
    <Unknown
      defaultActive={true}
      value={1}
    >
      Test2
    </Unknown>
    <Unknown
      value={2}
    >
      Test3
    </Unknown>
  </TabGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          isActive={false}
          selectTab={[Function]}
          value={0}
        >
          Test
        </Unknown>,
        <Unknown
          defaultActive={true}
          isActive={true}
          selectTab={[Function]}
          value={1}
        >
          Test2
        </Unknown>,
        <Unknown
          isActive={false}
          selectTab={[Function]}
          value={2}
        >
          Test3
        </Unknown>,
      ],
      "className": "tab-group-article",
      "style": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "function",
        "props": Object {
          "children": "Test",
          "isActive": false,
          "selectTab": [Function],
          "value": 0,
        },
        "ref": null,
        "rendered": "Test",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {
          "children": "Test2",
          "defaultActive": true,
          "isActive": true,
          "selectTab": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": "Test2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".2",
        "nodeType": "function",
        "props": Object {
          "children": "Test3",
          "isActive": false,
          "selectTab": [Function],
          "value": 2,
        },
        "ref": null,
        "rendered": "Test3",
        "type": [Function],
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            defaultActive={true}
            isActive={true}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={2}
          >
            Test3
          </Unknown>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": false,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "defaultActive": true,
            "isActive": true,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "function",
          "props": Object {
            "children": "Test3",
            "isActive": false,
            "selectTab": [Function],
            "value": 2,
          },
          "ref": null,
          "rendered": "Test3",
          "type": [Function],
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`TabGroup integration matchs base snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabGroup
    className={null}
    defaultValue={null}
    onChange={null}
    style={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "className": "tab-group-article",
      "style": null,
    },
    "ref": null,
    "rendered": null,
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": null,
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`TabGroup integration renders basic element if not a tab 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabGroup
    className={null}
    defaultValue={null}
    onChange={null}
    style={null}
  >
    <Unknown
      value={0}
    >
      Test
    </Unknown>
    <Unknown
      defaultActive={true}
      value={1}
    >
      Test2
    </Unknown>
    <span>
      This is not a tab
    </span>
  </TabGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          isActive={false}
          selectTab={[Function]}
          value={0}
        >
          Test
        </Unknown>,
        <Unknown
          defaultActive={true}
          isActive={true}
          selectTab={[Function]}
          value={1}
        >
          Test2
        </Unknown>,
        <span>
          This is not a tab
        </span>,
      ],
      "className": "tab-group-article",
      "style": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "function",
        "props": Object {
          "children": "Test",
          "isActive": false,
          "selectTab": [Function],
          "value": 0,
        },
        "ref": null,
        "rendered": "Test",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {
          "children": "Test2",
          "defaultActive": true,
          "isActive": true,
          "selectTab": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": "Test2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".2",
        "nodeType": "host",
        "props": Object {
          "children": "This is not a tab",
        },
        "ref": null,
        "rendered": "This is not a tab",
        "type": "span",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            isActive={false}
            selectTab={[Function]}
            value={0}
          >
            Test
          </Unknown>,
          <Unknown
            defaultActive={true}
            isActive={true}
            selectTab={[Function]}
            value={1}
          >
            Test2
          </Unknown>,
          <span>
            This is not a tab
          </span>,
        ],
        "className": "tab-group-article",
        "style": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "children": "Test",
            "isActive": false,
            "selectTab": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "children": "Test2",
            "defaultActive": true,
            "isActive": true,
            "selectTab": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": "Test2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".2",
          "nodeType": "host",
          "props": Object {
            "children": "This is not a tab",
          },
          "ref": null,
          "rendered": "This is not a tab",
          "type": "span",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
